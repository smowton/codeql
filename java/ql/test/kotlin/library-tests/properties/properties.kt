
abstract class properties(val constructorProp: Int, var mutableConstructorProp: Int, extractorArg: Int) {
    var modifiableInt = 1
    val immutableInt = 2
    val typedProp: Int = 3
    abstract val abstractTypeProp: Int
    val initialisedInInit: Int
    init {
        initialisedInInit = 4
    }
    val useConstructorArg = extractorArg
    val five: Int
            get() = 5
    val six
            get() = 6
    var getSet
            get() = modifiableInt
            set(v) { modifiableInt = v }
    val defaultGetter = 7
            get
    var varDefaultGetter = 8
            get
    var varDefaultSetter = 9
            set
    var varDefaultGetterSetter = 10
            get
            set
    var overrideGetter = 11
            get() = 12
    var overrideGetterUseField = 13
            get() = field + 1
    val useField = 14
            get() = field + 1
    lateinit var lateInitVar: String
    fun useProps(): Int {
        return 0 +
            constructorProp +
            mutableConstructorProp +
            modifiableInt +
            immutableInt +
            typedProp +
            abstractTypeProp +
            initialisedInInit +
            useConstructorArg +
            five +
            six +
            getSet +
            defaultGetter +
            varDefaultGetter +
            varDefaultSetter +
            varDefaultGetterSetter +
            overrideGetter +
            overrideGetterUseField +
            useField +
            constVal
    }
}

const val constVal = 15
