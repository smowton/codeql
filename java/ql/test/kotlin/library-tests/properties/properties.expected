| properties.kt:2:27:2:50 | constructorProp | properties.kt:2:27:2:50 | getConstructorProp | file://:0:0:0:0 | <none> | properties.kt:2:27:2:50 | constructorProp |
| properties.kt:2:53:2:83 | mutableConstructorProp | properties.kt:2:53:2:83 | getMutableConstructorProp | properties.kt:2:53:2:83 | setMutableConstructorProp | properties.kt:2:53:2:83 | mutableConstructorProp |
| properties.kt:3:5:3:25 | modifiableInt | properties.kt:3:5:3:25 | getModifiableInt | properties.kt:3:5:3:25 | setModifiableInt | properties.kt:3:5:3:25 | modifiableInt |
| properties.kt:4:5:4:24 | immutableInt | properties.kt:4:5:4:24 | getImmutableInt | file://:0:0:0:0 | <none> | properties.kt:4:5:4:24 | immutableInt |
| properties.kt:5:5:5:26 | typedProp | properties.kt:5:5:5:26 | getTypedProp | file://:0:0:0:0 | <none> | properties.kt:5:5:5:26 | typedProp |
| properties.kt:6:5:6:38 | abstractTypeProp | properties.kt:6:14:6:38 | getAbstractTypeProp | file://:0:0:0:0 | <none> | file://:0:0:0:0 | <none> |
| properties.kt:7:5:7:30 | initialisedInInit | properties.kt:7:5:7:30 | getInitialisedInInit | file://:0:0:0:0 | <none> | properties.kt:7:5:7:30 | initialisedInInit |
| properties.kt:11:5:11:40 | useConstructorArg | properties.kt:11:5:11:40 | getUseConstructorArg | file://:0:0:0:0 | <none> | properties.kt:11:5:11:40 | useConstructorArg |
| properties.kt:12:5:13:21 | five | properties.kt:13:13:13:21 | getFive | file://:0:0:0:0 | <none> | file://:0:0:0:0 | <none> |
| properties.kt:14:5:15:21 | six | properties.kt:15:13:15:21 | getSix | file://:0:0:0:0 | <none> | file://:0:0:0:0 | <none> |
| properties.kt:16:5:18:40 | getSet | properties.kt:17:13:17:33 | getGetSet | properties.kt:18:13:18:40 | setGetSet | file://:0:0:0:0 | <none> |
| properties.kt:19:5:20:15 | defaultGetter | properties.kt:20:13:20:15 | getDefaultGetter | file://:0:0:0:0 | <none> | properties.kt:19:5:20:15 | defaultGetter |
| properties.kt:21:5:22:15 | varDefaultGetter | properties.kt:22:13:22:15 | getVarDefaultGetter | properties.kt:21:5:22:15 | setVarDefaultGetter | properties.kt:21:5:22:15 | varDefaultGetter |
| properties.kt:23:5:24:15 | varDefaultSetter | properties.kt:23:5:24:15 | getVarDefaultSetter | properties.kt:24:13:24:15 | setVarDefaultSetter | properties.kt:23:5:24:15 | varDefaultSetter |
| properties.kt:25:5:27:15 | varDefaultGetterSetter | properties.kt:26:13:26:15 | getVarDefaultGetterSetter | properties.kt:27:13:27:15 | setVarDefaultGetterSetter | properties.kt:25:5:27:15 | varDefaultGetterSetter |
| properties.kt:28:5:29:22 | overrideGetter | properties.kt:29:13:29:22 | getOverrideGetter | properties.kt:28:5:29:22 | setOverrideGetter | properties.kt:28:5:29:22 | overrideGetter |
| properties.kt:30:5:31:29 | overrideGetterUseField | properties.kt:31:13:31:29 | getOverrideGetterUseField | properties.kt:30:5:31:29 | setOverrideGetterUseField | properties.kt:30:5:31:29 | overrideGetterUseField |
| properties.kt:32:5:33:29 | useField | properties.kt:33:13:33:29 | getUseField | file://:0:0:0:0 | <none> | properties.kt:32:5:33:29 | useField |
| properties.kt:34:5:34:36 | lateInitVar | properties.kt:34:14:34:36 | getLateInitVar | properties.kt:34:14:34:36 | setLateInitVar | properties.kt:34:5:34:36 | lateInitVar |
| properties.kt:59:1:59:23 | constVal | properties.kt:59:7:59:23 | getConstVal | file://:0:0:0:0 | <none> | properties.kt:59:1:59:23 | constVal |
